src/backend/distributed/README.md

```diff
@@ -0,0 +1,2079 @@
+
+## Principles 
+
+Use cases: 
+
+- Multi-tenant apps are the primary use case for Citus, which we can scale through distributing and co-locating by tenant ID, or through schema-based sharding. Citus is reasonably complete for this use case, but there are still SQL and operational improvements that can be made.
+- Real-time analytics is another popular use case due the combination of parallel distributed queries with indexes & in-database materialization (ETL). Improvement areas are automated time partitioning, better columnar storage (perf and update/delete, and incremental materialized views. 
+- Citus works well for CRUD use cases, but would be far easier to use if we introduced a load balancer, DDL from any node (no explicit coordinator), and more performant by better use of connection pooling (e.g. outbound pgbouncers). 
+- Marketplace use cases could work well if we made it easier to distribute tables twice by different dimensions or made it easier to keep paired tables in sync. 
+
+Schema management: 
+
+ - Our goal is for all DDL commands on Citus tables to work transparently, and for global DDL commands (e.g. CREATE TYPE) to be propagated to all nodes. Not all DDL is implemented yet and may either error or not propagate. 
+ - Since we cannot define custom DDL commands for sharding operations, we use functions that are called from a SELECT query. 
+
+Query layer:  
+
+- No incompatibilities with PostgreSQL – any query on a Citus table is supported on an equivalent PostgreSQL table. 
+- We optimize for maximum pushdown (& performance) over complete compatibility, but our long-term goal is for all queries to be supported in all cases. 
+- For single-shard queries, it is useful to avoid detailed query analysis through the fast path  planner (simple, single table) and router planner (co-located joins) layers. However, multi-shard queries can go through disparate code paths that were added out of expediency and should eventually be unified. 
+
+Transactional semantics: 
+
+- Transactions scoped to a single node follow the same semantics as PostgreSQL.  
+- Transactions across nodes are atomic, durable, and consistent, but do not have full snapshot isolation: A multi-shard query may see a concurrently committing transaction as committed on one node, but not yet committed on another node. 
+- Read-your-writes consistency should be preserved.  
+- Monotonic read consistency should be preserved for tables without replication, may not always be the case for replicated/reference tables. 
+
+Replication model: 
+
+- High availability is achieved through hot standby nodes. 
+- Read replicas are Citus cluster in which each node is a physical replica of a node in another Citus cluster. Since the read replica cannot have its own metadata. 
```

Comment:
The second sentence is not complete or does make sense on its own.

Wondering whether Read Replica definition should be explicitly added to the Concepts section. It's currently only discussed in the Cluster definition.